cool-demos/conditional-blocks.html

```diff
@@ -0,0 +1,452 @@
+
+                static get observedAttributes() { return ['data-property', 'data-operator', 'data-value']; }
+
+            }
+            // Register our web component to the tag we want to use it as
+            // https://developer.mozilla.org/en-US/docs/Web/API/CustomElementRegistry/define
+            win.customElements.define("conditional-block", ConditionalBlock);
+        }
+
+        // Custom function that manages the Insert/edit dialog
+        const dialogManager = (conditionalBlock, editor) => {
+            // Open a TinyMCE modal where the user can set the badge's
+            // background and text color.
+            // https://www.tiny.cloud/docs/ui-components/dialog/
+            // https://www.tiny.cloud/docs/ui-components/dialogcomponents/
+            editor.windowManager.open({
+                title: 'Insert/edit Conditional block',
+                body: {
+                    type: 'panel',
+                    items: [
+                        {
+                            type: 'selectbox',
+                            name: 'property',
+                            label: 'Property',
+                            items: [
+                                { value: 'number_of_people', text: 'number_of_people' },
+                                { value: 'name_of_event', text: 'name_of_event' },
+                                { value: 'length_of_stay', text: 'length_of_stay' },
+                            ]
+                        }, {
+                            type: 'selectbox',
+                            name: 'operator',
+                            label: 'Operator',
+                            items: [
+                                { value: 'is greater than', text: 'is greater than' },
+                                { value: 'is equal or greater than', text: 'is equal or greater than' },
+                                { value: 'is equal to', text: 'is equal to' },
+                                { value: 'is equal or less than', text: 'is equal or less than' },
+                                { value: 'is less than', text: 'is less than' },
+                                { value: 'is not equal to', text: 'is not equal to' },
+                            ]
+                        }, {
+                            type: 'input',
+                            name: 'value',
+                            label: 'Value'
+                        }
+                    ]
+                },
+                buttons: [
+                {
+                    type: 'cancel',
+                    name: 'closeButton',
+                    text: 'Cancel'
+                },
+                {
+                    type: 'submit',
+                    name: 'submitButton',
+                    text: 'Save',
+                    primary: true
+                }
+                ],
+                initialData: {
+                    property: conditionalBlock ? conditionalBlock.dataset.property : 'null' ,
```

Comment:
`'null'` isn't in your list of values available, so this doesn't make sense. We either need to make this match one of the existing values, or add a "none" version (eg `{ value: 'null', text: 'None' }`).